File: solutions/javascript/vehicle-purchase/2/vehicle-purchase.js
// @ts-check
//
// The line above enables type checking for this file. Various IDEs interpret
// the @ts-check directive. It will give you helpful autocompletion when
// implementing this exercise.

/**
 * Determines whether or not you need a license to operate a certain kind of vehicle.
 *
 * @param {string} kind
 * @returns {boolean} whether a license is required
 */
export function needsLicense(kind) {
  if( kind !== "car" && kind !== "truck")return false
  return true
}

/**
 * Helps choosing between two options by recommending the one that
 * comes first in dictionary order.
 *
 * @param {string} option1
 * @param {string} option2
 * @returns {string} a sentence of advice which option to choose
 */
export function chooseVehicle(option1, option2) {
 let result =  option1 < option2 ? option1 : option2;
  return  `${result} is clearly the better choice.`
}

/**
 * Calculates an estimate for the price of a used vehicle in the dealership
 * based on the original price and the age of the vehicle.
 *
 * @param {number} originalPrice
 * @param {number} age
 * @returns {number} expected resell price in the dealership
 */
export function calculateResellPrice(originalPrice, age) {
  const helper = (x, y) => (x / 100) * y;
  if (age < 3) {
    return helper(originalPrice, 80);
  }else if (age >= 3 && age <= 10) {
    return helper(originalPrice, 70);
  } else {
    return helper(originalPrice, 50);
  } 
}

File: solutions/go/vehicle-purchase/1/vehicle_purchase.go
package purchase
import "fmt"
// NeedsLicense determines whether a license is needed to drive a type of vehicle. Only "car" and "truck" require a license.
func NeedsLicense(kind string) bool {
	if kind == "car" || kind == "truck" {
        return true
    }
    return false
}

// ChooseVehicle recommends a vehicle for selection. It always recommends the vehicle that comes first in lexicographical order.
func ChooseVehicle(option1, option2 string) string {
	if option1 < option2 {
         return fmt.Sprintf( "%s is clearly the better choice.", option1)
    } 
    return fmt.Sprintf( "%s is clearly the better choice.", option2)
}

// CalculateResellPrice calculates how much a vehicle can resell for at a certain age.
func CalculateResellPrice(originalPrice, age float64) float64 {
	if age < 3 {
        return  originalPrice - discount(originalPrice,0.2)
    } else if age >= 3 && age < 10 {
        return  originalPrice - discount(originalPrice,0.3)
    } else{   
        return  originalPrice - discount(originalPrice,0.5)
    }
}

  func discount(price float64, discountPercent float64) float64 {
	return  price * discountPercent
    }